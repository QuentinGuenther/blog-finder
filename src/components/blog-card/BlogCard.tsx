import * as React from 'react';
import { BlogImage } from './BlogImage';
import { Interaction } from './Interaction';
import { Link } from './Link';

export interface BlogCardProps {
  imgPath: string,
  altText: string,
  title: string,
  domain: string,
  comments: number,
  votes: number,
  redditLink: string,
  blogLink: string
}

export function BlogCard (props: BlogCardProps) {
  return (
    <div>
      <BlogImage imgPath={props.imgPath} altText={props.altText} />
      <h2>{props.title}</h2>
      <a href={props.domain}>{props.domain}</a>
      <Interaction comments={props.comments} votes={props.votes} />
      <Link page={props.blogLink} children="Visit Page" />
      <Link page={props.redditLink} children="Comments" />
    </div>
  );
}
