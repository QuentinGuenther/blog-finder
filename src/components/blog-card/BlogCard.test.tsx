import * as React from 'react';
import renderer from 'react-test-renderer';
import { BlogCard } from './BlogCard';

describe('BlogCard should match snapshot', () => {
  it('BlogCard should match snapshot', () => {
    const tree = renderer
      .create(<BlogCard
        imgPath="https://b.thumbs.redditmedia.com/ZeHIfttfHHKwSTV3j87WqzzU6yMeCgm9gDTCcAKurIM.jpg"
        altText="A comprehensive list of … ES2017, ES2018, ES2019"
        title="A comprehensive list of … ES2017, ES2018, ES2019"
        domain="github.com"
        comments={3}
        votes={48}
        redditLink=	"https://reddit.com/r/webdev/comments/d8mj5o/a_comprehensive_list_of_new_es_features_including/"
        blogLink=	"https://github.com/daumann/ECMAScript-new-features-list"
      />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});