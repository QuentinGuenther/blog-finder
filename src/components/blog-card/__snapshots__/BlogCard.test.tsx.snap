// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogCard should match snapshot BlogCard should match snapshot 1`] = `
<div>
  <img
    alt="A comprehensive list of … ES2017, ES2018, ES2019"
    src="https://b.thumbs.redditmedia.com/ZeHIfttfHHKwSTV3j87WqzzU6yMeCgm9gDTCcAKurIM.jpg"
  />
  <h2>
    A comprehensive list of … ES2017, ES2018, ES2019
  </h2>
  <a
    href="github.com"
  >
    github.com
  </a>
  <div>
    <div
      className="comments"
    >
      3
    </div>
    <div
      className="vote"
    >
      48
    </div>
  </div>
  <a
    href="https://github.com/daumann/ECMAScript-new-features-list"
  >
    Visit Page
  </a>
  <a
    href="https://reddit.com/r/webdev/comments/d8mj5o/a_comprehensive_list_of_new_es_features_including/"
  >
    Comments
  </a>
</div>
`;
